### ISSUE TITLE
Startup error

### ISSUE THREAD

**Issue by refaelsh** [nvim-orgmode/orgmode#393]
### Describe the bug

Here is the error when I open and org file:

![2022-08-11_20-05](https://user-images.githubusercontent.com/2750775/184194392-31effab7-9275-4a8b-b649-ab2d8f531e60.png)

This does not happen with `minimal_init.lua`. It started happening a week or so ago.
I've must have done something to my init.lua.
Can you guys please take a look at the error and maybe point me to the problematic part of my init.lua?
Otherwise, I will start removing plugins one by one until this error goes away.

### Steps to reproduce

N/A

### Expected behavior

N/A

### Minimal init.lua

N/A

### OS / Distro

NixOS

### Neovim version/commit

0.7.2

**Comment by refaelsh** [comment]
On second thought, since it's not really a bug, maybe I should have posted in Discussions?

**Comment by kristijanhusak** [comment]
Did you try updating your tree-sitter grammar to latest version?
```
:TSUpdate org
```

**Comment by refaelsh** [comment]
> Did you try updating your tree-sitter grammar to latest version?
> 
> ```
> :TSUpdate org
> ```

Tried it just now. Did not help.

**Comment by kristijanhusak** [comment]
Can you try with the minimal init that should've been provided when opening the issue?

**Comment by refaelsh** [comment]
This *does not* happen with minimal_init.lua

**Comment by refaelsh** [comment]
This does not happen with minimal_init.lua. It started happening a week or so ago.
I've must have done something to my init.lua.
Can you guys please take a look at the error and maybe point me to the problematic part of my init.lua?
Otherwise, I will start removing plugins one by one until this error goes away.

**Comment by kristijanhusak** [comment]
From the error I can only see that tree-sitter is not able to parse something, but I'm not sure what, here and why. Are you sure you are using latest version of the plugin? I did some updates to the parsing last week to make it compatible with the latest version of the tree-sitter grammar, so you maybe have some discrepancy between orgmode and tree-sitter grammar version.

**Comment by refaelsh** [comment]
> Are you sure you are using latest version of the plugin?

Yes. I am using packer and I have configured it to always auto update packages.
So yes, I am using the latest versions of everything.

> latest version

Do you mean latest `master` or latest official release?

**Comment by kristijanhusak** [comment]
Latest master.
If yes, I would suggest to first try to remove plugins that are orgmode specific, like org bullets and such.

**Comment by refaelsh** [comment]
> Latest master.

I need to go and google to understand how packer works. I don't know if he uses the latest master or latest official release.

**Comment by refaelsh** [comment]
> I would suggest to first try to remove plugins that are orgmode specific, like org bullets and such.

Tried it, did not help.

**Comment by Ram-Z** [comment]
I'm experiencing this as well since updating orgmode to master just now.

**Comment by refaelsh** [comment]
> I'm experiencing this as well since updating orgmode to master just now.

Yes! I am not crazy! Yes!

**Comment by Ram-Z** [comment]
Bisected it to the following commit:

```
2022-08-05 1fecadc (refs/bisect/bad) [BREAKING CHANGE] Update tree-sitter grammar to latest version. Run :TSUpdate org
```

Ran `:TSUpdate org` and it resolved the issue. I cannot reproduce anymore.

**Comment by refaelsh** [comment]
Damn it :-) means I am still crazy :-) Running `:TSUpdate org` did not help me.

**Comment by kristijanhusak** [comment]
@refaelsh I think you will need to figure out which plugin causes the issue, since we cannot reproduce it.

**Comment by refaelsh** [comment]
> @refaelsh I think you will need to figure out which plugin causes the issue, since we cannot reproduce it.

Yea, I know. I am just lazy and keep putting this off... I will do it of course eventually and update here.

**Comment by refaelsh** [comment]
I found the problematic line in my `init.lua`.
I have this:
```lua
     require 'nvim-treesitter.configs'.setup {
         ensure_installed = {
             "all"
         },
         auto_install = true,
         highlight = {
             enable = true,
             additional_vim_regex_highlighting = { 'org' },
         },
         ensure_installed = { 'org' },
     }

```

If I remove the `enable = true` line, the error goes away.

A penny for you thoughts? :-)

**Comment by kristijanhusak** [comment]
You are just disabling TS highlights completely, that's why it goes away. At least we know it's something related to TS highlights. I think you will need to continue searching...

**Comment by refaelsh** [comment]
> You are just disabling TS highlights completely, that's why it goes away. At least we know it's something related to TS highlights. I think you will need to continue searching...

Understood. I will continue.

**Comment by refaelsh** [comment]
I got it down to this absolute minimum `init.lua` and the problem still happens.
```lua
local fn = vim.fn
local install_path = fn.stdpath('data') .. '/site/pack/packer/start/packer.nvim'
if fn.empty(fn.glob(install_path)) > 0 then
        packer_bootstrap = fn.system({ 'git', 'clone', '--depth', '1', 'https://github.com/wbthomason/packer.nvim',
        install_path })
        vim.cmd [[packadd packer.nvim]]
end
 
return require('packer').startup(function(use)
 
        use { "wbthomason/packer.nvim" }
 
        use 'nvim-treesitter/nvim-treesitter'
        use 'nvim-orgmode/orgmode'
 
        require('orgmode').setup_ts_grammar()
        require 'nvim-treesitter.configs'.setup {
                ensure_installed = {
                        "all"
                },
                auto_install = true,
                highlight = {
                        enable = true,
                        additional_vim_regex_highlighting = { 'org' },
                },
                ensure_installed = { 'org' },
        }
        require('orgmode').setup({
                org_agenda_files = { '~/Dropbox/org/*', '~/my-orgs/**/*' },
                org_default_notes_file = '~/Dropbox/org/refile.org',
        })
 
 
        vim.api.nvim_create_autocmd("BufWritePost", { pattern = "*.org", command = "!~/tangle.py %" })
        vim.api.nvim_create_autocmd("BufWritePost",
        { pattern = "configuration.org", command = "!sudo cp ~/configuration.nix /etc/nixos/configuration.nix" })
 
        vim.cmd([[
        augroup packer_user_config
        autocmd!
        autocmd BufWritePost plugins.lua source <afile> | PackerCompile
        augroup end
        ]])
 
        -- Automatically set up your configuration after cloning packer.nvim
        -- Put this at the end after all plugins
        if packer_bootstrap then
                require('packer').sync()
        end
end)    
```

Please suggest ideas.

**Comment by kristijanhusak** [comment]
Does it happen in each org file or only some specific one?
For example, does this error out:
```
* TODO Test
```

**Comment by refaelsh** [comment]
> Does it happen in each org file or only some specific one?

On each org file. Even an empty one.

For example, does this error out:
> 
> ```
> * TODO Test
> ```

Yes.

**Comment by kristijanhusak** [comment]
Try removing these autocmds, since it seems that's the only thing it's different from minimal init.

**Comment by refaelsh** [comment]
> Try removing these autocmds, since it seems that's the only thing it's different from minimal init.

Tried it. Did not help.

**Comment by kristijanhusak** [comment]
Send the output of `:scriptnames` after you open the file and it errors out

**Comment by refaelsh** [comment]
> Send the output of `:scriptnames` after you open the file and it errors out

```
:scriptnames                                                                                                                                                                          
  1: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/ftplugin.vim
  2: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/indent.vim
  3: ~/.config/nvim/init.lua
  4: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/filetype.lua
  5: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/filetype.vim
  6: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/syntax/syntax.vim
  7: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/syntax/synload.vim
  8: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/gzip.vim
  9: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/health.vim
 10: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/man.vim
 11: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/matchit.vim
 12: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
 13: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/matchparen.vim
 14: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/netrwPlugin.vim
 15: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/rplugin.vim
 16: /nix/store/q19zx4ck5hgb3ddsbk7jnmmbb576yll2-neovim-0.7.2/rplugin.vim
 17: ~/.local/share/nvim/rplugin.vim
 18: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/shada.vim
 19: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/spellfile.vim
 20: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/tarPlugin.vim
 21: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/tohtml.vim
 22: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/tutor.vim
 23: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/plugin/zipPlugin.vim
 24: ~/.config/nvim/plugin/packer_compiled.lua
 25: ~/.local/share/nvim/site/pack/packer/start/nvim-treesitter/plugin/nvim-treesitter.lua
 26: ~/.local/share/nvim/site/pack/packer/start/orgmode/plugin/orgmode.vim
 27: ~/.local/share/nvim/site/pack/packer/start/orgmode/ftplugin/org.vim
 28: ~/.local/share/nvim/site/pack/packer/start/orgmode/indent/org.vim
 29: ~/.local/share/nvim/site/pack/packer/start/orgmode/syntax/org.vim
 30: /nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime/syntax/lua.vim
```

**Comment by kristijanhusak** [comment]
Can you also try to remove path to your org files from the setup function and see if that helps?

**Comment by refaelsh** [comment]
> Can you also try to remove path to your org files from the setup function and see if that helps?

Do you mean delete these two  lines?

```lua
org_agenda_files = { '~/Dropbox/org/*', '~/my-orgs/**/*' },
org_default_notes_file = '~/Dropbox/org/refile.org',
```

**Comment by kristijanhusak** [comment]
Yeah, those.

**Comment by refaelsh** [comment]
> Can you also try to remove path to your org files from the setup function and see if that helps?

Tried it. Did not help.

**Comment by kristijanhusak** [comment]
Since you narrowed it down a lot, and minimal init from org repo is working for you, I would just suggest to try narrowing it down a bit more by comparing the two. Try using custom `rtp` and `packpath` like minimal init from here does and see if that helps.

**Comment by refaelsh** [comment]
> Since you narrowed it down a lot, and minimal init from org repo is working for you, I would just suggest to try narrowing it down a bit more by comparing the two. Try using custom `rtp` and `packpath` like minimal init from here does and see if that helps.

Nice suggestion. I will do it asap and report back here.

**Comment by refaelsh** [comment]
I've added the following lines:
```lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])
```
to my narrowed down init.lua and the error no longer happens.
I have no idea what those 2 lines do.

P.S.
Also, since I am on NixOS, and these 2 lines are doing something with some path or something, this might be a problem. NixOS is a very special distribution that does not confirm to the LFH standard.

**Comment by kristijanhusak** [comment]
These lines change the location of your runtime path and packpath. Basically the location of the plugins and everything else.
Seems like something is read from your rtp or packpath that causes the issue, but I didn't see anything strange in the `scriptnames`.
Can you remove these two lines, and give me the output of these?

```
:echo &runtimepath
```
and

```
:echo &packpath
```

**Comment by refaelsh** [comment]
> :echo &runtimepath

```
/home/refaelsh/.config/nvim,/etc/xdg/nvim,/home/refaelsh/.nix-profile/etc/xdg/nvim,/etc/profile
s/per-user/refaelsh/etc/xdg/nvim,/nix/var/nix/profiles/default/etc/xdg/nvim,/run/current-system
/sw/etc/xdg/nvim,/home/refaelsh/.local/share/nvim/site,/home/refaelsh/.local/share/nvim/site/pa
ck/packer/start/packer.nvim,/home/refaelsh/.local/share/nvim/site/pack/*/start/*,/nix/store/0k7
k3rpz0xigvb39kr5r57p90va4gw25-desktops/share/nvim/site,/home/refaelsh/.nix-profile/share/nvim/s
ite,/etc/profiles/per-user/refaelsh/share/nvim/site,/nix/var/nix/profiles/default/share/nvim/si
te,/run/current-system/sw/share/nvim/site,/nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-un
wrapped-0.7.2/share/nvim/runtime,/nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0
.7.2/share/nvim/runtime/pack/dist/opt/matchit,/nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovi
m-unwrapped-0.7.2/lib/nvim,/nix/store/nzkc4xlz432y1szi8c49fajvv9c9jz3y-vim-pack-dir/pack/*/star
t/*,/home/refaelsh/.local/share/nvim/site/pack/*/start/*/after,/run/current-system/sw/share/nvi
m/site/after,/nix/var/nix/profiles/default/share/nvim/site/after,/etc/profiles/per-user/refaels
h/share/nvim/site/after,/home/refaelsh/.nix-profile/share/nvim/site/after,/nix/store/0k7k3rpz0x
igvb39kr5r57p90va4gw25-desktops/share/nvim/site/after,/home/refaelsh/.local/share/nvim/site/aft
er,/run/current-system/sw/etc/xdg/nvim/after,/nix/var/nix/profiles/default/etc/xdg/nvim/after,/
etc/profiles/per-user/refaelsh/etc/xdg/nvim/after,/home/refaelsh/.nix-profile/etc/xdg/nvim/afte
r,/etc/xdg/nvim/after,/home/refaelsh/.config/nvim/after
Press ENTER or type command to continue
```

> :echo &packpath

```
/nix/store/nzkc4xlz432y1szi8c49fajvv9c9jz3y-vim-pack-dir,/home/refaelsh/.config/nvim,/etc/xdg/n
vim,/home/refaelsh/.nix-profile/etc/xdg/nvim,/etc/profiles/per-user/refaelsh/etc/xdg/nvim,/nix/
var/nix/profiles/default/etc/xdg/nvim,/run/current-system/sw/etc/xdg/nvim,/home/refaelsh/.local
/share/nvim/site,/nix/store/0k7k3rpz0xigvb39kr5r57p90va4gw25-desktops/share/nvim/site,/home/ref
aelsh/.nix-profile/share/nvim/site,/etc/profiles/per-user/refaelsh/share/nvim/site,/nix/var/nix
/profiles/default/share/nvim/site,/run/current-system/sw/share/nvim/site,/nix/store/p1a7kkcc893
nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime,/nix/store/p1a7kkcc893nhlqflwc1
x81xibrs4y8v-neovim-unwrapped-0.7.2/lib/nvim,/run/current-system/sw/share/nvim/site/after,/nix/
var/nix/profiles/default/share/nvim/site/after,/etc/profiles/per-user/refaelsh/share/nvim/site/
after,/home/refaelsh/.nix-profile/share/nvim/site/after,/nix/store/0k7k3rpz0xigvb39kr5r57p90va4
gw25-desktops/share/nvim/site/after,/home/refaelsh/.local/share/nvim/site/after,/run/current-sy
stem/sw/etc/xdg/nvim/after,/nix/var/nix/profiles/default/etc/xdg/nvim/after,/etc/profiles/per-u
ser/refaelsh/etc/xdg/nvim/after,/home/refaelsh/.nix-profile/etc/xdg/nvim/after,/etc/xdg/nvim/af
ter,/home/refaelsh/.config/nvim/after
Press ENTER or type command to continue
```

**Comment by kristijanhusak** [comment]
Seems there's a lot more things in both then usual, at least appears like that to me.
Would you consider installing Neovim through [bob](https://github.com/MordechaiHadad/bob)? I'm using it without any issues.

**Comment by refaelsh** [comment]
> Seems there's a lot more things in both then usual, at least appears like that to me. Would you consider installing Neovim through [bob](https://github.com/MordechaiHadad/bob)? I'm using it without any issues.

I would rather really not use `bob`. I am on NixOS and `bob` will preclude my from using `configuration.nix` for everything. Also, I think `bob` will never work on NixOS.

**Comment by kristijanhusak** [comment]
I'm not sure what you are talking about since I never used NixOS.
You will then have to get to the bottom of this. I would suggest to try commenting out rtp and leave the packpath, and vice versa. At least that will give you a source of the error (rtp or packpath).

**Comment by refaelsh** [comment]
> You will then have to get to the bottom of this.

Agreed.

> I would suggest to try commenting out rtp and leave the packpath, and vice versa. At least that will give you a source of the error (rtp or packpath).

I will try that.

**Comment by refaelsh** [comment]
I tried it only with this:
```lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
```

And the error went away.
I have no idea what this means.

**Comment by kristijanhusak** [comment]
I'm not sure that's a good idea. Can you give me the output of `echo $VIMRUNTIME` ?

**Comment by refaelsh** [comment]
> echo $VIMRUNTIME

```
/nix/store/p1a7kkcc893nhlqflwc1x81xibrs4y8v-neovim-unwrapped-0.7.2/share/nvim/runtime          
Press ENTER or type command to continue     
```

**Comment by kristijanhusak** [comment]
If you are using packer, I think things might work out. I would suggest to give it a try and see if it works. I don't have any other idea what can be wrong, but it's something from the paths in the rtp.

**Comment by refaelsh** [comment]
> If you are using packer, I think things might work out.

I do indeed use `packer` in my original huge init.lua.

>I don't have any other idea what can be wrong

Thank you anyway :-)

> but it's something from the paths in the rtp

I will try googling it. It might give me ideas.

**Comment by refaelsh** [comment]
I am still googling, but I get the feeling that it is something NixOS related. Just a feeling, nothing concrete to based it on.
In the meantime, I just now noticed that `akinsho/org-bullets.nvim` stopped working - I don't see a nice Unicode character, I see `*`s. Not really sure when it stopped working, but I noticed it just right now.

**Comment by refaelsh** [comment]
I discovered this thing called `:checkhealth nvim-treesitter`, and there is the result. Maybe it will shed some light.

![2022-08-25_19-54](https://user-images.githubusercontent.com/2750775/186725127-1fbc94e5-8da2-4815-9388-c0fd82925e1b.png)

**Comment by kristijanhusak** [comment]
It looks like your tree-sitter parser is loaded from a wrong location. Like you doing `:TSUpdate` doesn't really update the parsers.

**Comment by refaelsh** [comment]
> It looks like your tree-sitter parser is loaded from a wrong location. Like you doing `:TSUpdate` doesn't really update the parsers.

I have no idea what you just said. Can you please elaborate how you came to the conclusion of `It looks like your tree-sitter parser is loaded from a wrong location`.

**Comment by kristijanhusak** [comment]
I misread the path, I thought org parser is read from wrong location, but it seems fine. Can you try fully reinstalling `nvim-treesitter`? Just do `rm -rf ~/.local/share/nvim/site/pack/packer/start/nvim-treesitter` and then open up nvim and to a `PackerInstall`, or whichever is exactly the command.

**Comment by refaelsh** [comment]
> I misread the path, I thought org parser is read from wrong location, but it seems fine. Can you try fully reinstalling `nvim-treesitter`? Just do `rm -rf ~/.local/share/nvim/site/pack/packer/start/nvim-treesitter` and then open up nvim and to a `PackerInstall`, or whichever is exactly the command.

It did not help.

**Comment by refaelsh** [comment]
Solved! Thank you very much :-) The error no longer happens. The solution was found here: https://www.reddit.com/r/neovim/comments/oa3xq8/how_to_install_nvimtreesitter_on_nixos/. Search for an answer titled `Working solutions for NixO`.

TL;DR:
I delete stuff like this:
```
rm -rf ~/.local/share/nvim/site/pack/packer/start/nvim-treesitter
rm -rf ~/.local/share/nvim/site/parser
```

And reinstall all the plugins with `:PackerSync`.

**Comment by refaelsh** [comment]
It was indeed a NixOS related issue.

**Comment by refaelsh** [comment]
And the bullets now work too.

**Comment by kristijanhusak** [comment]
So it turns out parsers are really installed in a different location :)

I'm glad you solved it :)

**Comment by refaelsh** [comment]
> So it turns out parsers are really installed in a different location :)

Yes. Don't know why.

> I'm glad you solved it :)

Thank you again :-)
